### ISSUE TITLE
Error in Reshape - Conversion of VGG19 to TensorRT Engine - "reshaping failed for tensor: convolution_output"

### ISSUE THREAD

**Issue by sandeepganage** [onnx/onnx-tensorrt#588]
I am trying to convert a VGG19 segmentation network (ONNX) to TensorRT engine. 

Following is the network architecture : 

![ONNX_Model](https://user-images.githubusercontent.com/2736219/101380664-cf30b080-38db-11eb-847e-784a539a1bef.png)

I am running following command to convert the ONNX to tensorRT engine:
"trtexec.exe --onnx="onnx-model-path\model.onnx" --saveEngine="out-engine-path\Model.engine" --explicitBatch"

When I try to perform inference on the engine then it gives me following error:
![Console_Error](https://user-images.githubusercontent.com/2736219/101381332-a230cd80-38dc-11eb-80d9-dc60440e04db.PNG)

I can post the inference C++ code as well if it needs. It would be great if anyone can help me out with the issue.
Thanks in advance.

**Comment by pranavm-nvidia** [comment]
@sandeepganage Can you provide verbose logs as well? i.e. add `--verbose` to your trtexec command-line

**Comment by sandeepganage** [comment]
@pranavm-nvidia 
I regenerated engine with --verbose option. No error/warning found in log while building engine. After engine built, it goes for inference test. There I found following error:

```
[12/11/2020-16:17:02] [V] [TRT] Allocated persistent device memory of size 1070592
[12/11/2020-16:17:02] [V] [TRT] Allocated activation device memory of size 8535552
[12/11/2020-16:17:02] [V] [TRT] Assigning persistent memory blocks for various profiles
[12/11/2020-16:17:02] [E] [TRT] reshape_1: reshaping failed for tensor: convolution_output
[12/11/2020-16:17:02] [E] [TRT] C:\source\rtExt\shapeMachine.cpp (160) - Shape Error in nvinfer1::rt::`anonymous-namespace'::executeReshape: reshape would change volume
[12/11/2020-16:17:02] [E] [TRT] Instruction: RESHAPE_ZERO_IS_PLACEHOLDER{1 2 1 1} {0 2 262144}
[12/11/2020-16:17:02] [E] [TRT] reshape_1: reshaping failed for tensor: convolution_output
[12/11/2020-16:17:02] [E] [TRT] C:\source\rtExt\shapeMachine.cpp (160) - Shape Error in nvinfer1::rt::`anonymous-namespace'::executeReshape: reshape would change volume
[12/11/2020-16:17:02] [E] [TRT] Instruction: RESHAPE_ZERO_IS_PLACEHOLDER{1 2 1 1} {0 2 262144}
[12/11/2020-16:17:02] [E] [TRT] reshape_1: reshaping failed for tensor: convolution_output
[12/11/2020-16:17:02] [E] [TRT] C:\source\rtExt\shapeMachine.cpp (160) - Shape Error in nvinfer1::rt::`anonymous-namespace'::executeReshape: reshape would change volume
[12/11/2020-16:17:02] [E] [TRT] Instruction: RESHAPE_ZERO_IS_PLACEHOLDER{1 2 1 1} {0 2 262144}
[12/11/2020-16:17:02] [I] Starting inference
[12/11/2020-16:17:02] [E] [TRT] reshape_1: reshaping failed for tensor: convolution_output
[12/11/2020-16:17:02] [E] [TRT] C:\source\rtExt\shapeMachine.cpp (160) - Shape Error in nvinfer1::rt::`anonymous-namespace'::executeReshape: reshape would change volume
[12/11/2020-16:17:02] [E] [TRT] Instruction: RESHAPE_ZERO_IS_PLACEHOLDER{1 2 1 1} {0 2 262144}
[12/11/2020-16:17:02] [E] [TRT] reshape_1: reshaping failed for tensor: convolution_output
[12/11/2020-16:17:02] [E] [TRT] C:\source\rtExt\shapeMachine.cpp (160) - Shape Error in nvinfer1::rt::`anonymous-namespace'::executeReshape: reshape would change volume
[12/11/2020-16:17:02] [E] [TRT] Instruction: RESHAPE_ZERO_IS_PLACEHOLDER{1 2 1 1} {0 2 262144}
[12/11/2020-16:17:02] [E] [TRT] reshape_1: reshaping failed for tensor: convolution_output
[12/11/2020-16:17:02] [E] [TRT] C:\source\rtExt\shapeMachine.cpp (160) - Shape Error in nvinfer1::rt::`anonymous-namespace'::executeReshape: reshape would change volume
[12/11/2020-16:17:02] [E] [TRT] Instruction: RESHAPE_ZERO_IS_PLACEHOLDER{1 2 1 1} {0 2 262144}
[12/11/2020-16:17:02] [E] [TRT] reshape_1: reshaping failed for tensor: convolution_output
[12/11/2020-16:17:02] [E] [TRT] C:\source\rtExt\shapeMachine.cpp (160) - Shape Error in nvinfer1::rt::`anonymous-namespace'::executeReshape: reshape would change volume
[12/11/2020-16:17:02] [E] [TRT] Instruction: RESHAPE_ZERO_IS_PLACEHOLDER{1 2 1 1} {0 2 262144}
[12/11/2020-1
```

And engine generation It ends with this:

```
[12/11/2020-16:17:05] [I] Average on 10 runs - GPU latency: 10.607 ms - Host latency: 10.6424 ms (end to end 10.7108 ms, enqueue 10.5634 ms)
[12/11/2020-16:17:05] [I] Average on 10 runs - GPU latency: 10.5113 ms - Host latency: 10.5524 ms (end to end 10.607 ms, enqueue 10.4876 ms)
[12/11/2020-16:17:05] [I] Average on 10 runs - GPU latency: 10.4453 ms - Host latency: 10.4801 ms (end to end 10.5294 ms, enqueue 10.4023 ms)
[12/11/2020-16:17:05] [I] Average on 10 runs - GPU latency: 10.658 ms - Host latency: 10.7009 ms (end to end 10.7663 ms, enqueue 10.6221 ms)
[12/11/2020-16:17:05] [I] Average on 10 runs - GPU latency: 10.5915 ms - Host latency: 10.6429 ms (end to end 10.6988 ms, enqueue 10.5497 ms)
[12/11/2020-16:17:05] [I] Average on 10 runs - GPU latency: 10.5798 ms - Host latency: 10.6195 ms (end to end 10.6775 ms, enqueue 10.5391 ms)
[12/11/2020-16:17:05] [I] Average on 10 runs - GPU latency: 10.559 ms - Host latency: 10.6072 ms (end to end 10.6616 ms, enqueue 10.5276 ms)
[12/11/2020-16:17:05] [I] Average on 10 runs - GPU latency: 10.6802 ms - Host latency: 10.7221 ms (end to end 10.7866 ms, enqueue 10.6292 ms)
[12/11/2020-16:17:05] [I] Average on 10 runs - GPU latency: 10.5735 ms - Host latency: 10.6217 ms (end to end 10.6838 ms, enqueue 10.5327 ms)
[12/11/2020-16:17:05] [I] Average on 10 runs - GPU latency: 10.54 ms - Host latency: 10.5895 ms (end to end 10.6487 ms, enqueue 10.5014 ms)
[12/11/2020-16:17:05] [I] Average on 10 runs - GPU latency: 10.5462 ms - Host latency: 10.5915 ms (end to end 10.6404 ms, enqueue 10.5069 ms)
[12/11/2020-16:17:05] [I] Average on 10 runs - GPU latency: 10.5408 ms - Host latency: 10.5795 ms (end to end 10.6337 ms, enqueue 10.5033 ms)
[12/11/2020-16:17:05] [I] Average on 10 runs - GPU latency: 10.5487 ms - Host latency: 10.5864 ms (end to end 10.6372 ms, enqueue 10.4957 ms)
[12/11/2020-16:17:05] [I] Average on 10 runs - GPU latency: 10.5893 ms - Host latency: 10.6419 ms (end to end 10.6999 ms, enqueue 10.5533 ms)
[12/11/2020-16:17:05] [I] Average on 10 runs - GPU latency: 10.6116 ms - Host latency: 10.6559 ms (end to end 10.7141 ms, enqueue 10.5721 ms)
[12/11/2020-16:17:05] [I] Average on 10 runs - GPU latency: 10.5777 ms - Host latency: 10.6217 ms (end to end 10.683 ms, enqueue 10.5418 ms)
[12/11/2020-16:17:05] [I] Host Latency
[12/11/2020-16:17:05] [I] min: 10.1555 ms (end to end 10.218 ms)
[12/11/2020-16:17:05] [I] max: 11.3759 ms (end to end 11.4256 ms)
[12/11/2020-16:17:05] [I] mean: 10.6217 ms (end to end 10.68 ms)
[12/11/2020-16:17:05] [I] median: 10.6207 ms (end to end 10.6809 ms)
[12/11/2020-16:17:05] [I] percentile: 11.1752 ms at 99% (end to end 11.2262 ms at 99%)
[12/11/2020-16:17:05] [I] throughput: 0 qps
[12/11/2020-16:17:05] [I] walltime: 3.02375 s
[12/11/2020-16:17:05] [I] Enqueue Time
[12/11/2020-16:17:05] [I] min: 10.0566 ms
[12/11/2020-16:17:05] [I] max: 11.2731 ms
[12/11/2020-16:17:05] [I] median: 10.5374 ms
[12/11/2020-16:17:05] [I] GPU Compute
[12/11/2020-16:17:05] [I] min: 10.1138 ms
[12/11/2020-16:17:05] [I] max: 11.3188 ms
[12/11/2020-16:17:05] [I] mean: 10.5786 ms
[12/11/2020-16:17:05] [I] median: 10.5795 ms
[12/11/2020-16:17:05] [I] percentile: 11.0899 ms at 99%
[12/11/2020-16:17:05] [I] total compute time: 2.97259 s
&&&& PASSED TensorRT.trtexec # C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.1\bin\trtexec.exe --onnx=D:\TensorRt\artefact_model_10x\model\ONNX_Model\LiverArtefact_TF19_updatedWithBlurredTiles28112017.onnx --saveEngine=D:\TensorRt\artefact_model_10x\model\TRT_Model\Model_plane.engine --explicitBatch --verbose
```

Verbose output is pretty big to post here. So posting just the error and final part. Please let me know if you need anything else

**Comment by sandeepganage** [comment]
@pranavm-nvidia 

Some more details:
```
Input filename:   D:\TensorRt\artefact_model_10x\model\ONNX_Model\LiverArtefact_TF19_updatedWithBlurredTiles28112017.onnx
ONNX IR version:  0.0.6
Opset version:    11
Producer name:    keras2onnx
Producer version: 1.7.0
Domain:           onnxmltools
Model version:    0
```

nvcc.exe version output:
```
Copyright (c) 2005-2020 NVIDIA Corporation
Built on Tue_Sep_15_19:12:04_Pacific_Daylight_Time_2020
Cuda compilation tools, release 11.1, V11.1.74
Build cuda_11.1.relgpu_drvr455TC455_06.29069683_0
```

Cudnn version : 8.0.4

TensorRT version : 7.2.1.6

**Comment by pranavm-nvidia** [comment]
@sandeepganage It looks like it's actually able to run inference, which is strange. 

Regarding the error itself, are you sure the reshape is valid? What's the expected shape of the output of the preceding Conv? You could try using [Polygraphy](https://github.com/NVIDIA/TensorRT/tree/master/tools/Polygraphy) to run the model with ONNX-Runtime to double check that it's valid:
```
polygraphy run LiverArtefact_TF19_updatedWithBlurredTiles28112017.onnx --onnxrt
```

Otherwise, is it possible for you to share the model?

**Comment by sandeepganage** [comment]
@pranavm-nvidia Thanks for your reply. 
I will check the model with Polgraphy.
Meanwhile, following is the link to the ONNX model.

[](https://drive.google.com/file/d/14hSaXe6N4cUhm-XxkcyG-ENlVt_zTxdU/view?usp=sharing)

**Comment by pranavm-nvidia** [comment]
@sandeepganage It does look like a model issue after all; ONNX-Runtime reports something similar:
```
onnxruntime.capi.onnxruntime_pybind11_state.RuntimeException: [ONNXRuntimeError] : 6 : RUNTIME_EXCEPTION : Non-zero status code returned while running Reshape node. Name:'reshape_1' Status Message: /onnxruntime_src/onnxruntime/core/providers/cpu/tensor/reshape_helper.h:42 onnxruntime::ReshapeHelper::ReshapeHelper(const onnxruntime::TensorShape&, std::vector<long int>&) gsl::narrow_cast<int64_t>(input_shape.Size()) == size was false. The input tensor cannot be reshaped to the requested shape. Input shape:{1,2,1,1}, requested shape:{1,2,262144}
```

Can you try using `tf2onnx` to export your model rather than `keras2onnx`? I've seen similar bugs with `keras2onnx` in the past that didn't show up in `tf2onnx`

**Comment by sandeepganage** [comment]
@pranavm-nvidia : Hello Pranav.
Issue is still there.

============================================================

I ran following command to convert tf model to onnx:
`python -m tf2onnx.convert --saved-model "saved-model-path/" --output "path-output-model.onnx" --opset 12 --verbose`

============================================================

It gave me following output:
12:27:27,777 - INFO - tf2onnx.tfonnx: Using tensorflow=2.1.0, onnx=1.7.0, tf2onnx=1.7.1/796841
12:27:27,777 - INFO - tf2onnx.tfonnx: Using opset <onnx, 12>
12:28:20,327 - INFO - tf2onnx.tf_utils: Computed 0 values for constant folding
12:31:49,238 - VERBOSE - tf2onnx.tfonnx: Mapping TF node to ONNX node(s)
12:31:49,318 - VERBOSE - tf2onnx.tfonnx: Summay Stats:
        tensorflow ops: Counter({'Const': 136, 'Conv2D': 21, 'BiasAdd': 21, 'Pad': 20, 'Relu': 16, 'Transpose': 9, 'Shape': 5, 'StridedSlice': 5, 'Identity': 4, 'MaxPool': 4, 'Mul': 4, 'ResizeNearestNeighbor': 4, 'NoOp': 2, 'Placeholder': 1, 'Pack': 1, 'Reshape': 1, 'Max': 1, 'Sub': 1, 'Exp': 1, 'Sum': 1, 'RealDiv': 1})
        tensorflow attr: Counter({'dtype': 137, 'value': 136, 'T': 120, 'data_format': 46, 'strides': 25, 'padding': 25, 'dilations': 21, 'use_cudnn_on_gpu': 21, 'explicit_paddings': 21, 'Tpaddings': 20, 'Tperm': 9, 'out_type': 5, 'Index': 5, 'shrink_axis_mask': 5, 'ellipsis_mask': 5, 'begin_mask': 5, 'new_axis_mask': 5, 'end_mask': 5, 'ksize': 4, 'align_corners': 4, 'half_pixel_centers': 4, 'keep_dims': 2, 'Tidx': 2, 'shape': 1, 'axis': 1, 'N': 1, 'Tshape': 1})
        onnx mapped: Counter({'Const': 74, 'BiasAdd': 20, 'Relu': 16, 'Transpose': 9, 'Shape': 5, 'StridedSlice': 5, 'Identity': 4, 'MaxPool': 4, 'Mul': 4, 'ResizeNearestNeighbor': 4, 'Placeholder': 1, 'Conv2D': 1, 'Pack': 1, 'Reshape': 1, 'Max': 1, 'Sub': 1, 'Exp': 1, 'Sum': 1, 'RealDiv': 1})
        onnx unmapped: Counter()
12:31:49,321 - INFO - tf2onnx.optimizer: Optimizing ONNX model
12:31:49,322 - VERBOSE - tf2onnx.optimizer: Apply optimize_transpose
12:31:49,418 - VERBOSE - tf2onnx.optimizer.TransposeOptimizer: Const -46 (125->79), Reshape -20 (21->1), Transpose -16 (17->1)
12:31:49,419 - VERBOSE - tf2onnx.optimizer: Apply remove_redundant_upsample
12:31:49,439 - VERBOSE - tf2onnx.optimizer.UpsampleOptimizer: no change
12:31:49,439 - VERBOSE - tf2onnx.optimizer: Apply fold_constants
12:31:49,460 - VERBOSE - tf2onnx.optimizer.ConstFoldOptimizer: Unsqueeze -2 (3->1)
12:31:49,461 - VERBOSE - tf2onnx.optimizer: Apply loop_optimizer
12:31:49,479 - VERBOSE - tf2onnx.optimizer.LoopOptimizer: no change
12:31:49,479 - VERBOSE - tf2onnx.optimizer: Apply merge_duplication
12:31:49,540 - VERBOSE - tf2onnx.optimizer.MergeDuplicatedNodesOptimizer: Const -29 (79->50)
12:31:49,541 - VERBOSE - tf2onnx.optimizer: Apply remove_identity
12:31:49,816 - VERBOSE - tf2onnx.optimizer.IdentityOptimizer: Identity -5 (5->0)
12:31:49,817 - VERBOSE - tf2onnx.optimizer: Apply remove_back_to_back
12:31:49,840 - VERBOSE - tf2onnx.optimizer.BackToBackOptimizer: Squeeze -1 (1->0), Unsqueeze -1 (1->0)
12:31:49,840 - VERBOSE - tf2onnx.optimizer: Apply optimize_transpose
12:31:49,864 - VERBOSE - tf2onnx.optimizer.TransposeOptimizer: no change
12:31:49,864 - VERBOSE - tf2onnx.optimizer: Apply remove_redundant_upsample
12:31:49,881 - VERBOSE - tf2onnx.optimizer.UpsampleOptimizer: no change
12:31:49,881 - VERBOSE - tf2onnx.optimizer: Apply fold_constants
12:31:49,898 - VERBOSE - tf2onnx.optimizer.ConstFoldOptimizer: no change
12:31:49,898 - VERBOSE - tf2onnx.optimizer: Apply loop_optimizer
12:31:49,915 - VERBOSE - tf2onnx.optimizer.LoopOptimizer: no change
12:31:49,915 - VERBOSE - tf2onnx.optimizer: Apply merge_duplication
12:31:49,932 - VERBOSE - tf2onnx.optimizer.MergeDuplicatedNodesOptimizer: no change
12:31:49,932 - VERBOSE - tf2onnx.optimizer: Apply remove_identity
12:31:49,949 - VERBOSE - tf2onnx.optimizer.IdentityOptimizer: no change
12:31:49,949 - VERBOSE - tf2onnx.optimizer: Apply remove_back_to_back
12:31:49,966 - VERBOSE - tf2onnx.optimizer.BackToBackOptimizer: no change
12:31:49,968 - INFO - tf2onnx.optimizer: After optimization: Const -75 (125->50), Identity -5 (5->0), Reshape -20 (21->1), Squeeze -1 (1->0), Transpose -16 (17->1), Unsqueeze -3 (3->0)
12:31:49,980 - INFO - tf2onnx:
12:31:49,980 - INFO - tf2onnx: Successfully converted TensorFlow model D://TensorRt//artefact_model_10x//model//pb_model// to ONNX
12:31:50,775 - INFO - tf2onnx: ONNX model is saved at D://TensorRt//artefact_model_10x//model//ONNX_Model//model_opset12.onnx

============================================================

Everything looked good until I ran the `trtexec` to generate tensorRT engine:
[12:15:24] [I] Engine built in 21.4258 sec.
[12:15:24] [V] [TRT] Allocated persistent device memory of size 1070592
[12:15:24] [V] [TRT] Allocated activation device memory of size 10633728
[12:15:24] [V] [TRT] Assigning persistent memory blocks for various profiles
[12:15:24] [E] [TRT] StatefulPartitionedCall/sequential_1/reshape_1/Reshape: reshaping failed for tensor: StatefulPartitionedCall/sequential_1/conv2d_41/BiasAdd:0
[12:15:24] [E] [TRT] C:\source\rtExt\shapeMachine.cpp (160) - Shape Error in nvinfer1::rt::`anonymous-namespace'::executeReshape: reshape would change volume
[12:15:24] [E] [TRT] Instruction: RESHAPE_ZERO_IS_PLACEHOLDER{1 2 1 1} {1 2 262144}
[12:15:24] [E] [TRT] StatefulPartitionedCall/sequential_1/reshape_1/Reshape: reshaping failed for tensor: StatefulPartitionedCall/sequential_1/conv2d_41/BiasAdd:0
[12:15:24] [E] [TRT] C:\source\rtExt\shapeMachine.cpp (160) - Shape Error in nvinfer1::rt::`anonymous-namespace'::executeReshape: reshape would change volume
[12:15:24] [E] [TRT] Instruction: RESHAPE_ZERO_IS_PLACEHOLDER{1 2 1 1} {1 2 262144}
[12:15:24] [E] [TRT] StatefulPartitionedCall/sequential_1/reshape_1/Reshape: reshaping failed for tensor: StatefulPartitionedCall/sequential_1/conv2d_41/BiasAdd:0
[12:15:24] [E] [TRT] C:\source\rtExt\shapeMachine.cpp (160) - Shape Error in nvinfer1::rt::`anonymous-namespace'::executeReshape: reshape would change volume
[12:15:24] [E] [TRT] Instruction: RESHAPE_ZERO_IS_PLACEHOLDER{1 2 1 1} {1 2 262144}
[12:15:24] [I] Starting inference
[12:15:24] [E] [TRT] StatefulPartitionedCall/sequential_1/reshape_1/Reshape: reshaping failed for tensor: StatefulPartitionedCall/sequential_1/conv2d_41/BiasAdd:0
.....
.....
.....
[12:15:27] [E] [TRT] C:\source\rtExt\shapeMachine.cpp (160) - Shape Error in nvinfer1::rt::`anonymous-namespace'::executeReshape: reshape would change volume
[12:15:27] [E] [TRT] Instruction: RESHAPE_ZERO_IS_PLACEHOLDER{1 2 1 1} {1 2 262144}
[12:15:27] [I] Warmup completed 0 queries over 200 ms
[12:15:27] [I] Timing trace has 0 queries over 3.01794 s
[12:15:27] [I] Trace averages of 10 runs:
[12:15:27] [I] Average on 10 runs - GPU latency: 13.0521 ms - Host latency: 13.0917 ms (end to end 13.1538 ms, enqueue 12.9954 ms)
[12:15:27] [I] Average on 10 runs - GPU latency: 13.029 ms - Host latency: 13.0801 ms (end to end 13.1425 ms, enqueue 12.9916 ms)
[12:15:28] [I] Average on 10 runs - GPU latency: 13.1875 ms - Host latency: 13.2357 ms (end to end 13.3059 ms, enqueue 13.137 ms)
[12:15:28] [I] Average on 10 runs - GPU latency: 13.3038 ms - Host latency: 13.3396 ms (end to end 13.402 ms, enqueue 13.2493 ms)
[12:15:28] [I] Average on 10 runs - GPU latency: 13.1988 ms - Host latency: 13.2585 ms (end to end 13.3239 ms, enqueue 13.1516 ms)
[12:15:28] [I] Average on 10 runs - GPU latency: 13.0985 ms - Host latency: 13.1461 ms (end to end 13.2 ms, enqueue 13.0505 ms)
[12:15:28] [I] Average on 10 runs - GPU latency: 13.1161 ms - Host latency: 13.1701 ms (end to end 13.2286 ms, enqueue 13.0624 ms)
[12:15:28] [I] Average on 10 runs - GPU latency: 13.1418 ms - Host latency: 13.2038 ms (end to end 13.2672 ms, enqueue 13.0776 ms)
[12:15:28] [I] Average on 10 runs - GPU latency: 13.1791 ms - Host latency: 13.2468 ms (end to end 13.3183 ms, enqueue 13.1162 ms)
[12:15:28] [I] Average on 10 runs - GPU latency: 13.1656 ms - Host latency: 13.2225 ms (end to end 13.2837 ms, enqueue 13.1109 ms)
[12:15:28] [I] Average on 10 runs - GPU latency: 13.0417 ms - Host latency: 13.0833 ms (end to end 13.1332 ms, enqueue 12.9986 ms)
[12:15:28] [I] Average on 10 runs - GPU latency: 13.0505 ms - Host latency: 13.0798 ms (end to end 13.1401 ms, enqueue 13.0056 ms)
[12:15:28] [I] Average on 10 runs - GPU latency: 13.1655 ms - Host latency: 13.2074 ms (end to end 13.2582 ms, enqueue 13.1146 ms)
[12:15:28] [I] Average on 10 runs - GPU latency: 13.1765 ms - Host latency: 13.2072 ms (end to end 13.2761 ms, enqueue 13.1262 ms)
[12:15:28] [I] Average on 10 runs - GPU latency: 13.1417 ms - Host latency: 13.213 ms (end to end 13.285 ms, enqueue 13.0948 ms)
[12:15:28] [I] Average on 10 runs - GPU latency: 13.15 ms - Host latency: 13.2029 ms (end to end 13.2796 ms, enqueue 13.0942 ms)
[12:15:28] [I] Average on 10 runs - GPU latency: 13.1601 ms - Host latency: 13.2137 ms (end to end 13.2796 ms, enqueue 13.1094 ms)
[12:15:28] [I] Average on 10 runs - GPU latency: 13.0856 ms - Host latency: 13.137 ms (end to end 13.1915 ms, enqueue 13.0331 ms)
[12:15:28] [I] Average on 10 runs - GPU latency: 13.2231 ms - Host latency: 13.275 ms (end to end 13.3372 ms, enqueue 13.1677 ms)
[12:15:28] [I] Average on 10 runs - GPU latency: 13.1157 ms - Host latency: 13.1676 ms (end to end 13.2279 ms, enqueue 13.0589 ms)
[12:15:28] [I] Average on 10 runs - GPU latency: 13.1341 ms - Host latency: 13.188 ms (end to end 13.2396 ms, enqueue 13.0834 ms)
[12:15:28] [I] Average on 10 runs - GPU latency: 13.1774 ms - Host latency: 13.241 ms (end to end 13.3011 ms, enqueue 13.1206 ms)
[12:15:28] [I] Host Latency
[12:15:28] [I] min: 12.6311 ms (end to end 12.6689 ms)
[12:15:28] [I] max: 13.9139 ms (end to end 14.0032 ms)
[12:15:28] [I] mean: 13.1901 ms (end to end 13.2523 ms)
[12:15:28] [I] median: 13.183 ms (end to end 13.2383 ms)
[12:15:28] [I] percentile: 13.8271 ms at 99% (end to end 13.8812 ms at 99%)
[12:15:28] [I] throughput: 0 qps
[12:15:28] [I] walltime: 3.01794 s
[12:15:28] [I] Enqueue Time
[12:15:28] [I] min: 12.5488 ms
[12:15:28] [I] max: 13.7494 ms
[12:15:28] [I] median: 13.0797 ms
[12:15:28] [I] GPU Compute
[12:15:28] [I] min: 12.6053 ms
[12:15:28] [I] max: 13.8533 ms
[12:15:28] [I] mean: 13.1394 ms
[12:15:28] [I] median: 13.1288 ms
[12:15:28] [I] percentile: 13.7648 ms at 99%
[12:15:28] [I] total compute time: 2.96951 s
&&&& PASSED TensorRT.trtexec # C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.1\bin\trtexec.exe --onnx=D:\TensorRt\artefact_model_10x\model\ONNX_Model\model.onnx --saveEngine=D:\TensorRt\artefact_model_10x\model\TRT_Model\Model_tf2ONNX_opset12.engine --explicitBatch --verbose

============================================================

This is the second tensorflow model I tried converting to TensorRT engine. But no success. First one was with Unet architecture. This one is VGG-19

This is the link to my tensorflow model. If anyone wish to attempt the conversion then feel free to do so:
https://drive.google.com/drive/folders/1NCvpEMzX8S13rNrtjcglPQ8ng2e3WKoc?usp=sharing

============================================================

**Comment by sandeepganage** [comment]
@pranavm-nvidia  I have tested this model on 512 X 512 dimension tiles. Inference works absolutely fine and generates 512 X 512 output. So model works just fine.
From day 1 I have been facing this issue with tensorflow models. With caffe models (Resnet50) experience was better. This is the 3rd network I am trying to port to TensorRT (from tensorflow to tensorRT).

**Comment by pranavm-nvidia** [comment]
It definitely looks to me like a bug in `tf2onnx`/`keras2onnx`. You could try using [TF-TRT](https://docs.nvidia.com/deeplearning/frameworks/tf-trt-user-guide/index.html#worflow-with-savedmodel) instead.

**Comment by sandeepganage** [comment]
@pranavm-nvidia 
What is the most ideal way to run tensorflow model on tensorrt.? I tried with caffe model and it was easy as a snap.
I have been struggling to run tensorflow model using tensorrt from so many days. Now I am not sure if I am doing it the right way

**Comment by pranavm-nvidia** [comment]
@sandeepganage TF-TRT is definitely the easiest way to start. It functions just like any other TF optimization pass, and the resulting model can be used just like a typical TF model. 

On the other hand, importing into TRT via ONNX decouples you from the TF runtime completely and can give better performance in many cases. There are a lot of moving parts though, so it is more error-prone. Having said that, it does work very well for a large range of models, but it looks like you've found an edge case.

**Comment by sandeepganage** [comment]
Issue solved. Problem was with tensorflow model itself. I am using tensorflow2x and code which I was using to patch my model.h5 weight was in tensorflow1x. I was taking incorrect stride value for each convolution operation which ultimately resulted in giving me incorrect model itself. 
I converted this incorrect model to onnx and then was trying to deploy on tensorrt.
Bow the trt engine can be generated without any errors. But I it is still giving me incorrect reults. I will open another issue for that. 
Closing this issue. Thanks everyone. I really appreciate your valuable time and help.
